cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java

```diff
@@ -61,13 +65,41 @@ public final void append(ILoggingEvent eventObject) {
       if (loggingContext == null) {
         return;
       }
+      addErrorCodeTags(eventObject);
       addExtraTags(eventObject, loggingContext);
     }
 
     LogMessage logMessage = new LogMessage(eventObject, loggingContext);
     appendEvent(logMessage);
   }
 
+  /**
+   * if event is for an exception deriving from ErrorCodeProvider, capture error group etc.
+   * @param event
+   */
+  private void addErrorCodeTags(ILoggingEvent event) {
+    if (event.getThrowableProxy() != null && event.getThrowableProxy() instanceof ThrowableProxy) {
+      Throwable throwable = ((ThrowableProxy) (event.getThrowableProxy())).getThrowable();
+      // add all error tags from all the inner exceptions
+      // TODO We may also want to do some autodetection centrally, e.g. SocketException indicates it's network
```

Comment:
any todo should have an associate jira which gets referenced in the comment, otherwise it will never happen.

Reply:
Removed the TODO. This is just thinking aloud. Based on patterns of how we add tags, I am sure we can think of the best ways to do them in a clean and concise way.